xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/contentheader.vm

```diff
@@ -36,7 +36,8 @@
             $xwiki.formatDate($tdoc.date)
           ])
         #else
-          $services.localization.render('core.footer.modification', [$xwiki.getUserName($tdoc.author), $xwiki.formatDate($tdoc.date)])
+          #set ($authorName = $services.user.serialize($tdoc.authors.displayedAuthor))
```

Comment:
Wouldn't passing directly `$tdoc.authors.displayedAuthor` to getUserName have the same effect ? I did not check but I think we have a Converter for AuthorReference.

Reply:
> I did not check but I think we have a Converter for AuthorReference.

I guess you meant `UserReference`. We might have, not sure, but in general I prefer to do the conversion explicitely when it's possible. Do you feel it makes the code harder to read?

Reply:
Yes sorry `UserReference` and yes the point was just to suggest a simpler code.

Reply:
> Wouldn't passing directly $tdoc.authors.displayedAuthor to getUserName have the same effect ? I did not check but I think we have a Converter for AuthorReference.

So for info it does not because `XWiki#getUserName` takes a String as parameter, and the toString of the UserReference is not calling  the UserReference serializer, so we don't produce the proper output. So basically I do need to call the serializer before calling `XWiki#getUserName`

Reply:
> toString of the UserReference is not calling the UserReference serializer

That's not what is called during a conversion, it should go through UserReferenceConverter#convertToString.